syncd/syncd.cpp

```diff
@@ -2880,8 +2898,17 @@ void processFlexCounterEvent(
 
     sai_object_id_t vid = SAI_NULL_OBJECT_ID;
     sai_deserialize_object_id(vidStr, vid);
-    sai_object_id_t rid = translate_vid_to_rid(vid);
-    sai_object_type_t objectType = sai_object_type_query(rid);
+    sai_object_id_t rid;
+
+    if (!try_translate_vid_to_rid(vid, rid))
+    {
+        SWSS_LOG_WARN("port VID %s, was not found (probably port was removed/splitted) and will remove from counters now", 
+                sai_serialize_object_id(vid).c_str());
+
+        op = DEL_COMMAND;
+    }
```

Comment:
will there be another case, where flex counter request come first and by that time, the port vid has not been added to the mapping table, which end up with the case, the flex counter is not added at all?

Reply:
yes, this is possible, but no harm here, since FlexCounter::removeX() functios checks internally wheter vid is in the map, and only if then remove it

Reply:
I think the original question remains unanswered:  "which end up with the case, the flex counter is `not added at all`?"   

Or we treat it as a compromise?  Maybe it is time to unify flexcounter DB channel with asicDB channel to avoid the race condition?

Reply:
I ran into an issue that caused Orchagent to crash as follows:
1. swss added an ACL rule, and an associated new flex counter was passed to syncd to be added.
2. Flex counter somehow moved faster than the ACL rule, so sycnd decided that instead of adding this flex counter without finding the ACL rule sitting in the database, syncd did nothing, which is the intended behavior of this code change. 
3. Orchagent finished adding this ACL rule and thought that everything is fine.
4. Later on orchagent tried to look up the flex counter value but syncd reported that there is nothing there.
5. Orchagent then did abort().